### ISSUE TITLE
Don't require setting provider if there is an `ex_doc` section?

### ISSUE THREAD

**Issue by tsloughter** [erlef/rebar3_hex#341]
Would be nice if there wasn't a requirement to include:

```
{hex, [
    {doc, #{provider => ex_doc}}
]}.
```

In `rebar.config` if there is already an `ex_doc` section.

**Comment by starbelly** [comment]
Agreed. The only rub I can see is if you're doing both edoc and ex_doc for some reason, but could just override the default behavior with a switch in that case.
